src/lib/event.ts

```diff
@@ -177,10 +178,11 @@ const getEventsByDate = async (startDate: number, endDate: number): Promise<Even
   const events: JsonEvent[] = (await getObjects(keys)).filter(Boolean);
   const cids = await getCidsByPids(events.map(event => event.pid));
 
-  return events.map(fixEvent).map((event, i) => ({
+  return Promise.all(events.map(fixEvent).map(async (event, i) => ({
```

Comment:
If you make that change to `getResponsesCount` then you can avoid doing a separate database call for each event.